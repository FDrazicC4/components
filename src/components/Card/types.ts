import React from 'react';


export interface Props {


    /**
     *
     */
    actions?: React.ReactNode,


    /**
     * Any valid JSX to be wrapped in the main CardContent
     */
    children?: React.ReactNode,


    /**
     * Component used for the root node of children
     */
    contentComponent?: React.ElementType,


    /**
     * Any valid JSX to put before children
     */
    customContentBefore?: React.ReactNode,


    /**
     * Any valid JSX to put after children
     */
    customContentAfter?: React.ReactNode,


    /**
     * Action to appear in the header
     */
    headerAction?: React.ReactNode,


    /**
     * Avatar to appear in the header
     */
    headerAvatar?: React.ReactNode,


    /**
     * The root element for the header
     */
    headerComponent?: React.ElementType,


    /**
     * Subtitle for the header
     */
    headerSubtitle?: React.ReactNode,


    /**
     * Title for the header
     */
    headerTitle?: React.ReactNode,


    /**
     * Determines if the card is raised higher than normal
     */
    isRaised?: boolean,


}
