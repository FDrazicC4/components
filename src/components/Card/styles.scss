@import "../../styles/utilities";


// ==============================================
// General
// ==============================================

.card {
    box-shadow: map-get($general, box-shadow);
}

.noBorder {
    box-shadow: none;
}



