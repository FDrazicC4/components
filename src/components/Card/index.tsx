import React from 'react';
import clsx from "clsx";
import styles from './styles.scss';
import { Props } from './types';
import CardBody from './components/CardBody';
import CardHeader from './components/CardHeader';
import CardTitle from './components/CardTitle';

/**
 * The `<Card />` component is a flexible and extensible content container.
 */
function Card({ noBorder, className, children, ...otherProps }: Props) {
    const classes = clsx(
        styles.card,
        { [styles.noBorder]: !!noBorder },
        className,
    );
    return (
        <div className={classes} {...otherProps}>
            {children}
        </div>
    );
}

Card.Body = CardBody;
Card.Header = CardHeader;
Card.Title = CardTitle;

export default Card;
