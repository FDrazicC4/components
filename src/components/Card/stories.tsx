import React from 'react';
import { boolean } from '@storybook/addon-knobs';
import Card, { CardMedia } from '@components/Card';
import Icon from '@components/Icon';
import Typography from '@components/Typography';


export default {
    Component: Card,
    title: 'Stable/Data Display/Card',
}


export const Sandbox = (): React.ReactElement => {
    const isRaised = boolean('isRaised', false);

    return (
        <div>
            <Card isRaised={isRaised}>
                Foo
            </Card>

            <Card
                headerAction={<Icon>check</Icon>}
                headerTitle="Title"
                headerSubtitle="Subtitle"
                headerAvatar={<Icon>business</Icon>}
                customContentBefore={
                    <CardMedia
                        component="img"
                        src="https://cdn.pixabay.com/photo/2015/02/24/15/41/dog-647528__340.jpg"
                    />
                }
            >
                <Typography>
                    Hello World
                </Typography>
            </Card>
        </div>
    );
};
