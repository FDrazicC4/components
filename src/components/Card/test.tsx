import React from 'react';
import renderer from 'react-test-renderer';
import Card from '@components/Card';


it('renders correctly', () => {
    const tree = renderer
        .create((
            <Card>Default</Card>
        ))
        .toJSON();
    expect(tree).toMatchSnapshot();
});
