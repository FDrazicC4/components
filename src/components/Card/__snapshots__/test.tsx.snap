// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root brew-Card MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="brew-Card__content"
  >
    Default
  </div>
</div>
`;
